### DISCUSSION TITLE
Wondering if direct upload is possible?

### DISCUSSION THREAD

**Discussion by tosbourn** [railsadminteam/rails_admin#3621] [Q&A] [answered]
Hey folks,

I've been looking through the wiki and the source code and I don't believe direct file upload comes out of the box with Rails Admin, but I was wondering if anyone in the community has implemented anything?

I have two projects I'd like to use it on, one uses Paperclip to handle file uploads, and the other Active Storage.

If there was enough interest I'd be happy to look at submitting a PR with what we come up with.

**Comment by airled** [comment]
Don't really understand what you mean by direct upload. Rails admin supports paperclip, carrierwave and active storage out of the box. Here is the wiki pages
[uploads](https://github.com/railsadminteam/rails_admin/wiki/File-upload)

**Comment by tosbourn** [comment]
@airled thanks, direct upload is the concept where the frontend sends the file directly to the online storage instead of going via a backend. This gives the backend less work to do, and can lead to a better experience for users.

It is [documented in the rails guides](https://edgeguides.rubyonrails.org/active_storage_overview.html#direct-uploads)

**Comment by airled** [comment]
@tosbourn Oh. Thank you, now I get it. Seems like direct upload is not supported out of the box. I found this gem for s3 [rails_admin_s3_file](https://github.com/rootstrap/rails_admin_s3_file) but I think you found it too. It's outdated but as far as I can see the implementation is not that hard. But yeah, it's better to have such functionality out of the box.

**Comment by frank657** [comment, marked as answer]
@tosbourn not sure if you already found it, but direct upload is possible. Check out #3296

**Reply by tosbourn** [reply]
Thank you! I hadn't looked at this again but this is super useful.
